Race/race.py

```diff
@@ -0,0 +1,75 @@
+import random
+import time
+
+
+class Animal:
+
+    def generatenumber(self):
+        num = random.randint(1, 10)
+        return num
+
+    def move(self):
+        return self.generatenumber()
```

Comment:
Let's make this function a virtual function and make it compulsory for derived classes to implement it on their own instead of repeating the same function twice.

Reply:
I.e: see how to make a base class method virtual in python.